### PR TITLE
feat: Gh20635 native auth kube

### PR DESCRIPTION
**Pull request by sparkoo**
### What does this PR do?
Make it possible to deploy on kubernetes with native authentication. You still need to explicitly patch CheCluster with `nativeUserMode: true`.

I would also welcome any testing on both kubernetes and openshift, with and without devworkspaces.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20635

### How to test this PR?

```bash
cat EOF << > /tmp/patch.yaml
spec:
  auth:
    nativeUserMode: true
EOF

chectl server:deploy \
     --installer operator \
     --platform minikube \
     --che-operator-image quay.io/mvala/che-operator:gh20635-nativeAuthKube \
     --che-operator-cr-patch-yaml /tmp/patch.yaml \
     --workspace-engine=dev-workspace
```

1. deploy ^
2. import dex certificate into the browser
3. open Che URL
4. login with `che@eclipse.org:admin` or `userN@che:password` where N = [1-5]
5. Che will use user identity to talk with kubernetes. You can check `RoleBindings` created in user's namespace. You can test by coping for example request to get devworkspaces from browser network traffic. It should not work when you remove the `RoleBinding`.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
Pls run `make update-resources -s` to update olm bundles and commit changes.

**Comment by sparkoo** [comment]
> Pls run `make update-resources -s` to update olm bundles and commit changes.

yes. I was struggling to setup everything to run that procedure. Should be updated now.

**Comment by tolusha** [comment]
There is an issue with provisioning proxy cluster https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/eclipse-che_che-operator/1185/pull-ci-eclipse-che-che-operator-main-v8-che-behind-proxy/1465713039448739840
Let's give one more try.

and happypath tests are not working. let's ignore them

**Review by tolusha — APPROVED** [review]
Pls merge

**Comment by l0rd** [comment]
FWIW I have this error when trying to run the chectl server:start as you specified:

image

And from the logs:

```
2021-12-02T00:30:41.495Z Error: Command server:deploy failed. Error log: /Users/mloriedo/Library/Caches/chectl/error.log.
2021-12-02T00:30:41.495Z     at newError (~/.local/share/chectl/client/0.0.20211130-next.9db7957/lib/util.js:199:19)
2021-12-02T00:30:41.495Z     at Object.wrapCommandError (~/.local/share/chectl/client/0.0.20211130-next.9db7957/lib/util.js:195:12)
2021-12-02T00:30:41.495Z     at Deploy.<anonymous> (~/.local/share/chectl/client/0.0.20211130-next.9db7957/lib/commands/server/deploy.js:230:35)
2021-12-02T00:30:41.495Z     at Generator.throw (<anonymous>)
2021-12-02T00:30:41.495Z     at rejected (~/.local/share/chectl/client/0.0.20211130-next.9db7957/node_modules/tslib/tslib.js:115:69)
2021-12-02T00:30:41.495Z Cause: Error: Command failed with exit code 10: minikube cp /var/folders/c1/w7k_h7b90sdf0b579v8nyt780000gn/T/dex-ca.crt /etc/ca-certificates/dex-ca.crt
2021-12-02T00:30:41.495Z X Exiting due to MK_COMMAND_RUNNER: sudo test -d /etc/ca-certificates && sudo scp -t /etc/ca-certificates && sudo touch -d "2021-12-02 01:30:41.263866972 +0100" /etc/ca-certificates/dex-ca.crt: Process exited with status 1
2021-12-02T00:30:41.495Z output:
2021-12-02T00:30:41.495Z *
2021-12-02T00:30:41.495Z ╭───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
2021-12-02T00:30:41.495Z │                                                                                                                       │
2021-12-02T00:30:41.495Z │    * If the above advice does not help, please let us know:                                                           │
2021-12-02T00:30:41.495Z │      https://github.com/kubernetes/minikube/issues/new/choose                                                         │
2021-12-02T00:30:41.495Z │                                                                                                                       │
2021-12-02T00:30:41.495Z │    * Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.                              │
2021-12-02T00:30:41.495Z │    * Please also attach the following file to the GitHub issue:                                                       │
2021-12-02T00:30:41.495Z │    * - /var/folders/c1/w7k_h7b90sdf0b579v8nyt780000gn/T/minikube_cp_4a5c502840dfe1532adea2a71aaa6da6bcd781be_0.log    │
2021-12-02T00:30:41.495Z │                                                                                                                       │
2021-12-02T00:30:41.495Z ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
2021-12-02T00:30:41.495Z     at makeError (~/.local/share/chectl/client/0.0.20211130-next.9db7957/node_modules/execa/lib/error.js:60:11)
2021-12-02T00:30:41.495Z     at handlePromise (~/.local/share/chectl/client/0.0.20211130-next.9db7957/node_modules/execa/index.js:118:26)
```

I am testing on: 
```
minikube version: v1.24.0
commit: 76b94fb3c4e8ac5062daf70d60cf03ddcc0a741b
```

**Comment by tolusha** [comment]
@l0rd 
It is minikube 1.24 specific problem
https://github.com/eclipse/che/issues/20797

File: controllers/devworkspace/solver/che_routing.go [hunk 1]
```diff
@@ -514,13 +514,10 @@ func addEndpointToTraefikConfig(componentName string, e dw.Endpoint, cfg *gatewa
 		100,
 		fmt.Sprintf("http://127.0.0.1:%d", e.TargetPort),
 		[]string{prefix})
-	if util.IsOpenShift4 {
-		cfg.AddAuth(name, fmt.Sprintf("http://%s.%s:8089?namespace=%s", gateway.GatewayServiceName, cheCluster.Namespace, routing.Namespace))
-	}
+	cfg.AddAuth(name, fmt.Sprintf("http://%s.%s:8089?namespace=%s", gateway.GatewayServiceName, cheCluster.Namespace, routing.Namespace))
```
Comment (metlos) [inline controllers/devworkspace/solver/che_routing.go:517]: This reminds me... Are we still meant to support OpenShift 3? If yes, how is this going to behave on OpenShift 3?

**Review by metlos — COMMENTED** [review]
LGTM but how is this behaving on OpenShift 3, if we still need to support it?

File: controllers/devworkspace/solver/che_routing.go (hunk 1 above)
Comment (sparkoo) [reply controllers/devworkspace/solver/che_routing.go:517]: TBH I don't know. But looking at current `che_routing.go`, I doubt it's working now.

**Pull request merged**